### PR TITLE
File 458 files in mongo

### PR DESCRIPTION
**Pull request by YuanZHANG910**
add method for retrieve Mongo Files, add one unit-test and one group of it-test

### CODE REVIEW

File: app/uk/gov/hmrc/fileupload/file/zip/Zippy.scala [hunk 1]
```diff
@@ -86,4 +98,31 @@ object Zippy {
     Enumerator(baos.toByteArray)
   }
 
+  //Todo: remove ALL following when mongoDB is not in use at all.
+
+  def checkIsTheFileInS3(fileRefId:FileRefId): Boolean = {
+    val mongoRegex = "([0-9a-f]{8})-([0-9a-f]{4})-([0-9a-f]{4})-([0-9a-f]{4})-([0-9a-f]{12})".r
```
Comment (codesurf42) [inline app/uk/gov/hmrc/fileupload/file/zip/Zippy.scala]: change to explicit dashes, ([0-9a-f]{8})\-([0-9a-f]{4})

File: app/uk/gov/hmrc/fileupload/file/zip/Zippy.scala [hunk 2]
```diff
@@ -86,4 +98,31 @@ object Zippy {
...
+    val s3Regex = """([\d\D.]{32})""".r
```
Comment (codesurf42) [inline app/uk/gov/hmrc/fileupload/file/zip/Zippy.scala]: I can't find such expectation about S3 ids, they can be longer

File: app/uk/gov/hmrc/fileupload/file/zip/Zippy.scala [hunk 3]
```diff
@@ -86,4 +98,31 @@ object Zippy {
...
+    fileRefId.value match {
+      case mongoRegex(_,_,_,_,_) => false
+      case s3Regex(_) => true
+      case _ => false
```
Comment (codesurf42) [inline app/uk/gov/hmrc/fileupload/file/zip/Zippy.scala]: remove _ case

File: it/uk/gov/hmrc/fileupload/DownloadEnvelopeIntegrationSpec.scala [hunk 4]
```diff
@@ -59,6 +60,48 @@ class DownloadEnvelopeIntegrationSpec extends IntegrationSpec with EnvelopeActio
...
+
+
+  feature("Download Envelope MongoDB") {
+
+    scenario("A client can download an envelope including its file") {
+      Given("I have an envelope with files")
+      val envelopeId = createEnvelope()
+      val data = "my file content"
+      val fileId = FileId(s"fileId-${nextId()}")
+      val fileRefId = FileRefId()
```
Comment (kastoestoramadus) [inline it/uk/gov/hmrc/fileupload/DownloadEnvelopeIntegrationSpec.scala]: Not related to this ticket but still valuable:
TODO: FileRedId should not have a default value anymore

File: it/uk/gov/hmrc/fileupload/support/EventsActions.scala [hunk 5]
```diff
@@ -32,4 +33,16 @@ trait EventsActions extends ActionsSupport {
       .url(s"$url/commands/store-file")
       .post(Json.toJson(e))
       .futureValue
+
+  def sendFileInQuarantineStored(e: FileInQuarantineStored): WSResponse =
```
Comment (kastoestoramadus) [inline it/uk/gov/hmrc/fileupload/support/EventsActions.scala]: TODO: annotate those are temparary

File: it/uk/gov/hmrc/fileupload/DownloadEnvelopeIntegrationSpec.scala [hunk 6]
```diff
@@ -59,6 +60,48 @@ class DownloadEnvelopeIntegrationSpec extends IntegrationSpec with EnvelopeActio
...
+      And("File has been stored in quarantine on the front-end")
+      sendFileInQuarantineStored(
+        FileInQuarantineStored(envelopeId, fileId, fileRefId, 0, "file-name", "contentType", Some(123L), Json.obj("metadata" -> "foo"))
+      )
+
+      And("File was scanned and no virus was found")
+      sendFileScanned(FileScanned(envelopeId, fileId, fileRefId, hasVirus = false))
+
+      And("File was stored in transient")
+      upload(data.getBytes, envelopeId, fileId, fileRefId)
```
Comment (kastoestoramadus) [inline it/uk/gov/hmrc/fileupload/DownloadEnvelopeIntegrationSpec.scala]: Isn't it a S3 upload? what is the point of this test?

**Comment by kastoestoramadus** [comment]
I would like to merge it. Any objections?

**Pull request merged**